linked_list/.~c9_invoke_17vQM9.py

```diff
@@ -0,0 +1,172 @@
+class AbstractLinkedList(object):
+    """
+    Abstract class representing the LinkedList inteface you must respect.
+    
+    You must not implement any of the method in this class, and this class
+    must never be instantiated. It's just a "guide" of which methods
+    the LinkedList class should respect.
+    """
+
+    def __str__(self):
+        raise NotImplementedError()
+
+    def __len__(self):
+        raise NotImplementedError()
+
+    def __iter__(self):
+        raise NotImplementedError()
+
+    def __getitem__(self, index):
+        raise NotImplementedError()
+
+    def __add__(self, other):
+        raise NotImplementedError()
+
+    def __iadd__(self, other):
+        raise NotImplementedError()
+
+    def __eq__(self, other):
+        raise NotImplementedError()
+
+    def append(self, element):
+        raise NotImplementedError()
+
+    def count(self):
+        raise NotImplementedError()
+
+    def pop(self, index=None):
+        raise NotImplementedError()
+
+
+class Node(object):
+    """
+    Node class representing each of the linked nodes in the list.
+    """
+
+    def __init__(self, elem, next=None):
+        self.elem = elem
+        self.next = next
+
+    def __str__(self):
+        return str(self.elem)
+
+    def __eq__(self, other):
+        if isinstance(other, LinkedList):
+            return self.elem == other.elem #compare node.elem to value passed to it
+        else:
+            return self.elem == other
+
+    def __repr__(self):
+        return "<Node {}>".format(self.elem)
+
+
+class LinkedList(AbstractLinkedList):
+    """
+    Implementation of an AbstractLinkedList inteface.
+    """
+
+    def __init__(self, elements=None):
+        # start at none
+        self.start = None
+        self.end = None
+        #[1,2], self.start (Node(1)) >> Node1.next now pts to Node(2) 
+        # set elements to empty list
+        elements = [] if not elements else elements
+        for val in elements:
+            self.append(val)
+            
+
+    def __str__(self):
+        return str([element.elem for element in self])
+    
+    def __len__(self):
+        return self.count()
+
+    def __iter__(self):
+        n = self.start
+        while n:
+            yield n
+            n = n.next
+
+    def __getitem__(self, index):
+       for n, node in enumerate(self): 
+           if n == index:
+               return node
+
+    def __add__(self, other):
+       l1 = [ n.elem for n in self]
+       l2 = [ n.elem for n in other]
+       l3 = l1+l2
+       return LinkedList(l3)
+
+    def __iadd__(self, other):
+        return self.__add__(other) #  x += 1  == x = x+1
```

Comment:
This creates a whole new linked list by using \_\_add\_\_. Instead you could just use your append() method:
```python
for el in other:
  append(el)
```
which I think would be more memory efficient (no new instance created) but possibly at the cost of speed because of the for loop.